### DISCUSSION TITLE
Power failure,  Thorium does not recover

### DISCUSSION THREAD

**Discussion by Adecabo9** [edrlab/thorium-reader#3040] [Developer corner]
A user of library:
_While loading Thorium Reader (TR), the power failed and I turned off my PC. When I restarted I couldn't activate TR, I uninstalled and reinstalled, but it still gives ERROR (attached image) 1st when executing and 2nd when 'Accept' Error-01 
My PC data: W 10 updated 
Chrome updated._

![thorium 1](https://github.com/edrlab/thorium-reader-doc/assets/118538546/5b93fb5e-3791-405e-a66e-092b1859e984)

![thorium 2](https://github.com/edrlab/thorium-reader-doc/assets/118538546/45ecef7f-c7de-46b8-92bf-28a0066a0b0c)

**Comment by danielweck** [comment]
Try to copy the folder `EDRLab.ThoriumReader` shown in your second screenshot (make a backup somewhere on your computer), then delete the original `C:/Users/ERUPE/AppData/Local/Programs/EDRLab.ThoriumReader`, then start Thorium again.
What version of Thorium is this (did you install it from the Windows Store?)

**Comment by Adecabo9** [comment]
The version is the last (2.4) and he installed it from page of Thorium: https://thorium.edrlab.org/

**Comment by Adecabo9** [comment]
The user continues with the same error. Can you help us please?

**Comment by llemeurfr** [comment]
The sudden power failure seems to have made at least one file unusable. 
Do you mean that the file that failed loading was deleted (along with its parent directory) and the app still does not start?

Warning : Before deleting the directory indicated by Daniel for testing purpose, it must be saved elsewhere. If not Thorium's ebook catalog will be lost. If the original ebook files are correctly saved or easy to download for the user, there is not issue deleting this Thorium catalog. The user will simply have to re-import the ebooks.

**Comment by ERP-2459** [comment]
I am the library user.

Last Friday I sent an email and received a response to do the same as they indicated in the first days of opening this matter.

In the new installation with Thorium 3.0.0.0, the file EDRLab.ThoriumReader/resources/... does not appear; and the error indicated above continues to appear.

![Directorio de instalación en C](https://github.com/edrlab/thorium-reader/assets/174267720/e19213df-d380-4caa-841a-2fde622d0d88)

(Shortly after starting this delivery of information to resolve it, I independently deleted ALL the records included in Windows that contained information from Thorium Reader)

**Comment by arthur-lemeur** [comment]
On windows I uninstalled Thorium 2.4 which deleted all the files in my "User/AppData/Local/Programs/EDRLab.ThoriumReader" folder. Then I installed Thorium 3.0, the files were not recreated in this folder, it remained empty, I did not find the new folder where they were created. 
I also made a global search for "Thorium.exe" and nothing came back.

**Comment by danielweck** [comment]
Did you uninstall the Windows Store version of Thorium, or the NSIS standalone installer?
If the latter, that is a worrying bug because the NSIS uninstaller is not supposed to remove config files, publication database, etc. If the former, this is an annoying behaviour that needs to be documented!

**Comment by danielweck** [comment]
follow-up:

https://www.electron.build/configuration/nsis.html

deleteAppDataOnUninstall = false Boolean - one-click installer only. Whether to delete app data on uninstall.

**Comment by danielweck** [comment]
Also note that since Thorium 3.0 the NSIS installer isn't automatic anymore ("one click"), it is an assisted installer.
https://github.com/edrlab/thorium-reader/blob/0edc73d1ccc252227215bdcf4f5ab6263c511930/package.json#L225

**Comment by ERP-2459** [comment]
After deleting EVERYTHING related to "Thorium" and "EDRLab" from the Windows Registry (these are the deleted registries):

![Registro de Windows (con Thorium) 01](https://github.com/edrlab/thorium-reader/assets/174267720/bbca85ab-4466-4e54-aba3-79a530fbb1a4)
![Registro de Windows (con Thorium) 02](https://github.com/edrlab/thorium-reader/assets/174267720/1b3b42dc-8fad-47cb-aee4-aefc434dd1b1)

Restarting PC, I reinstalled Thorium 3.0.0.0 giving the same errors:

![ERROR 01 TR 3 0 0 0](https://github.com/edrlab/thorium-reader/assets/174267720/c83a8418-1336-477b-b0ce-be284a1c79cb)
![ERROR 02 TR 3 0 0 0](https://github.com/edrlab/thorium-reader/assets/174267720/63d5a5b0-b6f4-4f21-895a-fc7bb2e4a7c5)

Am I doing something wrong or is it just really bad luck?

**Comment by danielweck** [comment]
is that Windows 10 or 11?

**Comment by ERP-2459** [comment]
Windows 10  x64
![Sistema Operativo - Windows](https://github.com/edrlab/thorium-reader/assets/174267720/aacc508c-9808-4152-bc86-44a2d652363f)

![Detalles OS](https://github.com/edrlab/thorium-reader/assets/174267720/94d03dc7-aa9a-4d30-8fe5-b77a26ffa9f4)

Web browser:
![Navegador - Chrome](https://github.com/edrlab/thorium-reader/assets/174267720/98fcdc06-5ee3-4195-8f43-08a3c269a719)

**Comment by danielweck** [comment]
Did you install Thorium from the Windows Store or from the standalone installer executable that is downloaded from the GitHub releases?

**Comment by ERP-2459** [comment]
https://thorium.edrlab.org/

![thorium Reader - EDRLAB](https://github.com/edrlab/thorium-reader/assets/174267720/cd7e6f30-d7fb-4d70-9ef4-7407aa5d8711)

**Comment by danielweck** [comment]
do you experience the same problem with a test / preview build from here:

https://github.com/edrlab/thorium-reader/releases/tag/latest-windows-intel

**Comment by ERP-2459** [comment]
With this new version (3.1.0) it runs perfectly. (Thank you very much)

![Thorium 3 1 0](https://github.com/edrlab/thorium-reader/assets/174267720/4450d406-e18b-4488-9884-0cf4ba3486a8)


[Although the access code (from my Library) to reading newspapers or magazines is not working and I cannot read the information]

Should I talk to them about my code or is this new version not yet authorized to download and read information?

**Comment by danielweck** [comment]
unfortunately the preview / test builds do not support LCP-protected publications, which I suspect is why you are not able to access the books in your library portal.

**Comment by danielweck** [comment]
suggestion: please uninstall the Thorium preview/test build (using the Windows operating system "remove programs" feature), and install the official Thorium v3 again (from the executable installer, not from the Windows Store)

**Comment by ERP-2459** [comment]
Uninstalling v3.1 as indicated and installing v3.0, unfortunately the same errors happen again:

![Error reinstalación TR 3 0 0 - 5 jul](https://github.com/edrlab/thorium-reader/assets/174267720/1244bf14-b846-4ef6-ac69-40140cfc8123)

![Error reinstalación TR 3 0 0 0 - 5 jul](https://github.com/edrlab/thorium-reader/assets/174267720/442ced97-dd52-4f83-9238-64bbc0843555)

I will have to be patient and wait for the final activation of v3.1...

**Comment by danielweck** [comment]
this is so strange / unexplained. there is no major difference between 3.0 official release and the latest test build.

**Comment by ERP-2459** [comment]
I know; I'm devastated but I can't find a solution to these errors.
